### DISCUSSION TITLE
Expand limit

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2450] [Q&A]
Is there any limit to expand records?

**Comment by ganigeorgiev** [comment]
No, but I'm planning to add in the near future. Initially it will have a fixed value of maybe 1k and later with the inverse filter will add a collection option to control the expand records limit and the allowed client-side expand relations (both direct and indirect).

**Reply by Oggy4th** [reply]
is it possible to use perPage in expand, I would like to only retrieve 10 of the latest expand

Edit: or maybe use filter in expand

**Reply by ganigeorgiev** [reply]
> is it possible to use perPage in expand
>  or maybe use filter in expand

No, the APIs are starting to become too complicated. If you want paginated expand results and filtering, it may better to just send a separate request.

**Reply by Oggy4th** [reply]
Ohh gotcha, thank you very much for a quick response. I'll send a separate request then.

**Comment by abrman** [comment]
Has there been any developments in terms of limiting expand? Fly.io just told me my machine ran out of memory and I have my suspicions :)

**Reply by ganigeorgiev** [reply]
Nothing has changed here. If you are on a lower spec vm and often get OOM errors, you can try to:

1. [Enable swap](https://fly.io/docs/rails/cookbooks/deploy/#enabling-swap)
    Create a helper `docker_cmd.sh` file with something like:
    ```sh
    # https://fly.io/docs/rails/cookbooks/deploy/#enabling-swap
    fallocate -l 512M /swapfile
    chmod 0600 /swapfile
    mkswap /swapfile
    echo 10 > /proc/sys/vm/swappiness
    swapon /swapfile
    echo 1 > /proc/sys/vm/overcommit_memory

    # run the application
    /pb/pocketbase serve --dir='/pb/pb_data' --http='0.0.0.0:8080'
    ```
    And then change the last `CMD` instruction in your Dockerfile to run the script instead:
    ```dockerfile
    COPY docker_cmd.sh /docker_cmd.sh

    CMD /docker_cmd.sh
    ```
    
2. Set the `GOMEMLIMIT` env variable (eg. `GOMEMLIMIT=200MiB`) in your Docker/fly.toml file.

**Reply by abrman** [reply]
Thank you kindly. - I won't bother with the details, but I've cleared out some of the unnecessary relations and it's been running smoothly. I'll circle back to here when I have more time to explore this solution. <3

**Reply by ganigeorgiev** [reply]
You can also try temporary disabling the logs by changing the _Admin UI > Application > "Logs max days retention"_ input to `0` as the logs write currently are not very optimized (_this will be fixed with the upcoming logs generalization in the next release_).
